### PR TITLE
Theme fixes and enhancements

### PR DESCRIPTION
**Pull request by DartVanya**
### [Full DarkMode support for TaskDialog](https://github.com/winsiderss/systeminformer/pull/2231/commits/fc395e6dd565351dfc046c0dc2e80bc335b703e2)
Credits to https://github.com/SFTRS/DarkTaskDialog

- Transit from PhShow(Error|Warning|Information) to PhShow(Error|Warning|Information)2.
- ExtendedServices: MessageBox to TaskDialog transit.
---
- Public PhThemeWindowDrawButton: support tristate checkbox, radio, multiline label.

- Apply DarkMode_Explorer theme for all checkboxes (even for ListView checks), radio and buttons.
- Fix artefacts on GroupBox and Static picture controls when window moving back from off-screen.
- TreeNew:  significantly improve visibility of selected item.
- ExtListVew: fix text readability for hot and selected colored items.
- ~~Fix drawing of disabled ListView in dark theme~~ Hide ListVIew on Graph settings page instead of disabling.
- Fix multiline edit artifacts when typing, fix scrollbar.
- Fix dark theme for new choose dialog.
- Fix dark ListView header in unsupported system dialogs (skip superclass drawing).
- Better ListView theme (DarkMode_ItemsView and OpenNcThemeData hook for scrollbars).
- Add borders to TabControl header and body, selected tab slightly enlarged (behaves like default MS white theme).
- Add separators for StatusBar parts for better readability.
- SysLink theme support.
- Security page CHECKLIST_ACLUI checks theme support and other improvements.
- Add new theme options: EnableThemeAnimation (default 1), EnableThemeNativeButtons (default 0).
---
- Advanced options: menu Reset added (support multiple selection).
- Highlighting and Graph colors: menu - Reset added.
---
All changes highlighted on video:

https://github.com/user-attachments/assets/211387c4-b26f-4822-b3ff-5b0c5ad4e9b1

https://github.com/user-attachments/assets/28c94cf2-eb1e-495a-bbe6-a734f5394a70

https://github.com/user-attachments/assets/38c12dee-bf5d-4d01-8c97-b258b45b2f71

https://github.com/user-attachments/assets/ba8a69f3-08cb-4cda-90a5-3bf2b5821394

### CODE REVIEW

File: SystemInformer/delayhook.c [hunk 1]
```diff
@@ -1451,33 +1714,359 @@ BOOL WINAPI PhSystemParametersInfoHook(
...
+        return PhDefaultOpenNcThemeData(NULL, L"Explorer::ScrollBar");
+    }
+
+    return PhDefaultOpenNcThemeData(hwnd, pszClassList);
+}
+
+BOOLEAN CALLBACK PhInitializeTaskDialogTheme(
+    _In_ HWND WindowHandle,
+    _In_opt_ PVOID CallbackData
+)
```
Comment (jxy-s) [inline SystemInformer/delayhook.c:1882]: nit: noticing a few areas where the `)` isn't indented - don't worry about correcting them unless you end up having to make changes - @dmex or I can go in and clean them up after merging

File: SystemInformer/main.c [hunk 2]
```diff
@@ -133,7 +133,10 @@ INT WINAPI wWinMain(
     }
 
     PhGuiSupportInitialization();
-    PhpInitializeSettings();
+    if (!PhStartupParameters.Debug)
+    {
+        PhpInitializeSettings();
+    }
```
Comment (jxy-s) [inline SystemInformer/main.c:139]: I don't think this change is necessary?

File: phlib/theme.c [hunk 3]
```diff
@@ -2487,6 +2611,46 @@ VOID ThemeWindowRenderTabControl(
...
+
+        if (TabCtrl_GetItem(WindowHandle, currentSelection, &tabItem))
+        {
+            DrawText(
+                bufferDc,
+                tabItem.pszText,
+                (UINT)PhCountStringZ(tabItem.pszText),
+                &itemRect,
+                DT_CENTER | DT_VCENTER | DT_SINGLELINE | DT_HIDEPREFIX
+            );
```
Comment (jxy-s) [inline phlib/theme.c:2634]: nit: number of places where `);` is not aligned property for a function call - again don't worry about it unless you end up having to make changes - can be cleaned up after merge by @dmex or I

**Review by jxy-s — APPROVED** [review]
Changes seem fine to me. I had some nits and a question but overall I'm okay with merging it. I would prefer to give @dmex and opportunity to review since this is touching some of the UI code I'm not all that familiar with.

**Comment by Vasilich** [comment]
@DartVanya 
If you already on the UI for dark mode - can you fix the dialogs for network plugin too? There are dialogs like Ping and traceroute, that look weitd...

**Comment by DartVanya** [comment]
> dialogs like Ping and traceroute

Yes, it already fixed. See after.mp4 near the end.

File: phlib/mapldr.c [hunk 4]
```diff
@@ -1712,8 +1712,9 @@ VOID PhLoaderEntrySnapShowErrorMessage(
 
         if (IMAGE_SNAP_BY_ORDINAL(OriginalThunk->u1.Ordinal))
         {
-            PhShowError(
+            PhShowError2(
                 NULL,
+                L"Error.",
```
Comment (dmex) [inline phlib/mapldr.c:1717]: Swap "Error" for PhApplicationName?

File: plugins/ExtendedTools/gpudetails.c [hunk 5]
```diff
@@ -577,7 +577,7 @@ VOID EtShowGpuDetailsDialog(
     {
         if (!NT_SUCCESS(PhCreateThreadEx(&EtGpuDetailsDialogThreadHandle, EtGpuDetailsDialogThreadStart, ParentWindowHandle)))
         {
-            PhShowError(NULL, L"%s", L"Unable to create the window.");
+            PhShowError2(GetDesktopWindow(), L"Unable to create the window.", L"%s", L"");
```
Comment (dmex) [inline plugins/ExtendedTools/gpudetails.c:580]: GetDesktopWindow?

File: plugins/ExtendedTools/npudetails.c [hunk 6]
```diff
@@ -579,7 +579,7 @@ VOID EtShowNpuDetailsDialog(
     {
         if (!NT_SUCCESS(PhCreateThreadEx(&EtNpuDetailsDialogThreadHandle, EtNpuDetailsDialogThreadStart, ParentWindowHandle)))
         {
-            PhShowError(NULL, L"%s", L"Unable to create the window.");
+            PhShowError2(GetDesktopWindow(), L"Unable to create the window.", L"%s", L"");
```
Comment (dmex) [inline plugins/ExtendedTools/npudetails.c:582]: GetDesktopWindow?

File: plugins/ToolStatus/main.c [hunk 7]
```diff
@@ -1413,7 +1413,7 @@ LRESULT CALLBACK MainWindowProc(
                                 }
                                 else
                                 {
-                                    PhShowError(WindowHandle, L"The process (PID %lu) does not exist.", processId);
+                                    PhShowError2(WindowHandle, L"Error.", L"The process (PID %lu) does not exist.", processId);
```
Comment (dmex) [inline plugins/ToolStatus/main.c:1416]: PhApplicationName?

File: plugins/ToolStatus/main.c [hunk 8]
```diff
@@ -1428,7 +1428,7 @@ LRESULT CALLBACK MainWindowProc(
...
```
Comment (dmex) [inline plugins/ToolStatus/main.c:1431]: PhApplicationName?

File: plugins/Updater/updater.c [hunk 9]
```diff
@@ -1377,7 +1377,7 @@ VOID ShowUpdateDialog(
     {
         if (!NT_SUCCESS(PhCreateThreadEx(&UpdateDialogThreadHandle, ShowUpdateDialogThread, Context)))
         {
-            PhShowError(NULL, L"%s", L"Unable to create the window.");
+            PhShowError2(GetDesktopWindow(), L"Unable to create the window.", L"%s", L"");
```
Comment (dmex) [inline plugins/Updater/updater.c:1380]: GetDesktopWindow?

File: tools/peview/main.c [hunk 10]
```diff
@@ -117,13 +117,18 @@ INT WINAPI wWinMain(
...
     if (PhIsExecutingInWow64())
     {
-        PhShowWarning(
-            NULL,
+        PhGuiSupportInitialization();
+        PhSettingsInitialization();
+        PvInitializeSettings();
+        PvInitializeSuperclassControls();
+        PhShowWarning2(
+            GetDesktopWindow(),
```
Comment (dmex) [inline tools/peview/main.c:125]: GetDesktopWindow?

File: tools/peview/main.c [hunk 11]
```diff
@@ -306,9 +311,9 @@ INT WINAPI wWinMain(
         if (!NT_SUCCESS(status))
         {
             if (status == STATUS_IMAGE_SUBSYSTEM_NOT_PRESENT)
-                PhShowError2(NULL, L"Unable to load the file.", L"%s", L"PE Viewer does not support this image type.");
+                PhShowError2(GetDesktopWindow(), L"Unable to load the file.", L"%s", L"PE Viewer does not support this image type.");
             else
-                PhShowStatus(NULL, L"Unable to load the file.", status, 0);
+                PhShowStatus(GetDesktopWindow(), L"Unable to load the file.", status, 0);
```
Comment (dmex) [inline tools/peview/main.c:316]: ^

File: tools/peview/main.c [hunk 12]
```diff
@@ -306,9 +311,9 @@ INT WINAPI wWinMain(
         if (!NT_SUCCESS(status))
         {
             if (status == STATUS_IMAGE_SUBSYSTEM_NOT_PRESENT)
-                PhShowError2(NULL, L"Unable to load the file.", L"%s", L"PE Viewer does not support this image type.");
+                PhShowError2(GetDesktopWindow(), L"Unable to load the file.", L"%s", L"PE Viewer does not support this image type.");
```
Comment (dmex) [inline tools/peview/main.c:314]: ^

File: plugins/UserNotes/options.c [hunk 13]
```diff
@@ -292,7 +292,7 @@ INT_PTR CALLBACK OptionsDlgProc(
 
                                     for (LONG i = numberOfItems - 1; i >= 0; i--)
                                     {
-                                        PhRemoveListViewItem(listviewHandle, i);
+                                        PhRemoveListViewItem(listviewHandle, PhFindListViewItemByParam(listviewHandle, INT_ERROR, listviewItems[i]));
```
Comment (dmex) [inline plugins/UserNotes/options.c:295]: this is searching for the index of an item when we already have the index?

**Pull request closed without merging**